plugins/essif-lab/src/Application/Workflows/ManageHooks.php

```diff
@@ -2,35 +2,46 @@
 
 namespace TNO\EssifLab\Application\Workflows;
 
+use Exception;
 use TNO\EssifLab\Contracts\Abstracts\Workflow;
+use TNO\EssifLab\Services\PostUtil;
 
 class ManageHooks extends Workflow {
-	public static function options(): array {
+    private const MAX_ID = "maxID";
+    private const CONTEXT = 'context';
+    private const TARGET = 'target';
+
+    public static function options(): array {
 		// TODO: load all selectable options to be displayed in the select lists.
-		return ['context' => ['hello' => 'hello', 'world' => 'world'], 'target' => ['foo' => 'foo', 'bar' => 'bar']];
+		return [self::CONTEXT => ['hello' => 'hello', 'world' => 'world'], self::TARGET => ['foo' => 'foo', 'bar' => 'bar']];
 	}
 
 	public function add($request) {
-		$data = $this->getJsonPostContentAsArray();
+		$data = PostUtil::getJsonPostContentAsArray();
 		$hooks = array_key_exists('hook', $data) ? $data['hook'] : null;
 
 		if ($this->requestIsEmpty($request)) {
 			// TODO: add custom exception (request is empty)
-			throw new \Exception('request is empty');
+			throw new Exception('request is empty');
 		}
 
-		if (! empty($hooks)) {
+		if (!empty($hooks)) {
 			if ($this->doesHookAlreadyExists($hooks, $request)) {
 				// TODO: add custom exception (request already exists)
-				throw new \Exception('request already exists');
+				throw new Exception('request already exists');
 			}
-			$hooks[] = $this->generateHookRecord($hooks, $request);
+			$request = $this->generateHookRecord($hooks, $request);
+            $hooks[self::MAX_ID] = $request["ID"];
```

Comment:
Als het goed is heeft `$request` nooit een `ID` bij een add actie. Ook snap ik niet waarom je een attribuut `MAX_ID` toevoegd aan de `$hooks` array. Nu krijg je de volgende JSON:

```
[
  'hooks' => [
    ['ID' => 1, 'context' => 'foo', 'target' => 'bar' ],
    'maxID' => null,
  ]
]
```